File: VB/GrowthWare.Framework/Model/Profiles/MGroupProfile.vb
Imports GrowthWare.Framework.Model.Profiles.Base
Imports GrowthWare.Framework.Model.Profiles.Interfaces

Namespace Model.Profiles
    ''' <summary>
    ''' Model object representing the GroupProfile
    ''' </summary>
    ''' <remarks></remarks>
    <Serializable(), CLSCompliant(True)>
    Public Class MGroupProfile
        Inherits MProfile

#Region "Member Properties"
        Private m_Description As String = String.Empty
        Private m_SecurityEntityId As Integer = 1
#End Region

#Region "Protected Methods"
        ''' <summary>
        ''' Initializes the properties using the specified DataRow.
        ''' </summary>
        ''' <param name="dataRow">The dr.</param>
        Protected Overloads Sub Initialize(ByVal dataRow As DataRow)
            Me.IdColumnName = "GROUP_SEQ_ID"
            Me.NameColumnName = "NAME"
            MyBase.Initialize(dataRow)
            m_Description = MyBase.GetString(dataRow, "DESCRIPTION")
        End Sub
#End Region

#Region "Public Methods"
        ''' <summary>
        ''' Will return a message profile with the default vaules
        ''' </summary>
        ''' <remarks></remarks>
        Public Sub New()

        End Sub

        ''' <summary>
        ''' Initializes a new instance of the <see cref="MGroupProfile" /> class.
        ''' </summary>
        ''' <param name="dataRow">The DataRow</param>
        Public Sub New(ByVal dataRow As DataRow)
            Me.Initialize(dataRow)
        End Sub
#End Region

#Region "Public Properties"
        ''' <summary>
        ''' Gets or sets the security entity ID.
        ''' </summary>
        ''' <value>The security entity ID.</value>
        Public Property SecurityEntityId() As Integer
            Get
                Return m_SecurityEntityId
            End Get
            Set(ByVal value As Integer)
                m_SecurityEntityId = value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the description.
        ''' </summary>
        ''' <value>The description.</value>
        Public Property Description() As String
            Get
                Return m_Description
            End Get
            Set(ByVal value As String)
                If Not String.IsNullOrEmpty(value) Then m_Description = value.Trim
            End Set
        End Property
#End Region
    End Class
End Namespace

File: CS/GrowthWare.Framework/Model/Profiles/MGroupProfile.cs
using GrowthWare.Framework.Model.Profiles.Base;
using GrowthWare.Framework.Model.Profiles.Interfaces;
using System;
using System.Data;

namespace GrowthWare.Framework.Model.Profiles
{
    /// <summary>
    /// Model object representing the GroupProfile
    /// </summary>
    [Serializable(), CLSCompliant(true)]
    public class MGroupProfile : MProfile
    {
    #region "Member Properties"
        private string m_Description = string.Empty;
        private int m_SecurityEntityId = 1;
    #endregion

    #region "Protected Methods"
        /// <summary>
        /// Initializes the specified DataRow.
        /// </summary>
        /// <param name="dataRow">The DataRow.</param>
        protected new void Initialize(DataRow dataRow)
        {
            base.NameColumnName = "NAME";
            base.IdColumnName = "GROUP_SEQ_ID";
            base.Initialize(dataRow);
            m_Description = base.GetString(dataRow, "DESCRIPTION");
        }
        #endregion

        #region "Public Methods"
        /// <summary>
        /// Will return a message profile with the default value's
        /// </summary>
        /// <remarks></remarks>
        public MGroupProfile()
        {
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="MGroupProfile" /> class.
        /// </summary>
        /// <param name="profileDataRow">The profile data row.</param>
        public MGroupProfile(DataRow profileDataRow)
        {
            this.Initialize(profileDataRow);
        }
        #endregion

        #region "Public Properties"
        /// <summary>
        /// Gets or sets the security entity ID.
        /// </summary>
        /// <value>The security entity ID.</value>
        public int SecurityEntityId
        {
            get { return m_SecurityEntityId; }
            set { m_SecurityEntityId = value; }
        }

        /// <summary>
        /// Gets or sets the description.
        /// </summary>
        /// <value>The description.</value>
        public string Description
        {
            get { return m_Description; }
            set { if(value != null) m_Description = value.Trim(); }
        }
        #endregion
    }
}
